bottlecap/src/otlp/transform.rs

```diff
@@ -0,0 +1,1144 @@
+    let messaging_operation =
+        get_otel_attribute_value_as_string(&otel_span.attributes, MESSAGING_OPERATION, true);
+    if !messaging_system.is_empty() && !messaging_operation.is_empty() {
+        match otel_span.kind() {
+            SpanKind::Client | SpanKind::Server | SpanKind::Consumer | SpanKind::Producer => {
+                return format!("{messaging_system}.{messaging_operation}");
+            }
+            _ => {}
+        }
+    }
+
+    // RPC & AWS
+    let rpc_system = get_otel_attribute_value_as_string(&otel_span.attributes, RPC_SYSTEM, true);
+    let is_rpc = !rpc_system.is_empty();
+    let is_aws = is_rpc && rpc_system == "aws-api";
+
+    // AWS client
+    if is_aws && is_client {
+        let service = get_otel_attribute_value_as_string(&otel_span.attributes, RPC_SERVICE, true);
+        if !service.is_empty() {
+            return format!("aws.{service}.request");
+        }
+
+        return "aws.client.request".to_string();
+    }
+
+    // RPC client
+    if is_rpc && is_client {
+        return format!("{rpc_system}.client.request");
+    }
+
+    // RPC server
+    if is_rpc && is_server {
+        return format!("{rpc_system}.server.request");
+    }
+
+    // FaaS client
+    let provider =
+        get_otel_attribute_value_as_string(&otel_span.attributes, FAAS_INVOKED_PROVIDER, true);
+    let invoked_name =
+        get_otel_attribute_value_as_string(&otel_span.attributes, FAAS_INVOKED_NAME, true);
+    if !provider.is_empty() && !invoked_name.is_empty() && is_client {
+        return format!("{provider}.{invoked_name}.invoke");
+    }
+
+    // FaaS server
+    let trigger = get_otel_attribute_value_as_string(&otel_span.attributes, FAAS_TRIGGER, true);
+    if !trigger.is_empty() && is_server {
+        return format!("{trigger}.invoke");
+    }
+
+    // GraphQL
+    if !get_otel_attribute_value_as_string(&otel_span.attributes, "graphql.operation.type", true)
+        .is_empty()
+    {
+        return "graphql.server.request".to_string();
+    }
+
+    // If nothing matches, checking for generic http server/client
+    let protocol =
+        get_otel_attribute_value_as_string(&otel_span.attributes, "network.protocol.name", true);
+    if is_server {
+        if !protocol.is_empty() {
+            return format!("{protocol}.server.request");
+        }
+        return "server.request".to_string();
+    } else if is_client {
+        if !protocol.is_empty() {
+            return format!("{protocol}.client.request");
+        }
+        return "client.request".to_string();
+    }
+
+    if otel_span.kind() != SpanKind::Unspecified {
+        return otel_span.kind().as_str_name().to_string();
+    }
+
+    SpanKind::Internal.as_str_name().to_string()
+}
+
+#[must_use]
+#[allow(clippy::too_many_lines)]
+pub fn get_otel_resource(otel_span: &OtelSpan, otel_res: &OtelResource) -> String {
+    let resource_name = get_otel_attribute_value_as_string_from_resource_or_span(
```

Comment:
I'm not sure if this is method is best done as a series of cascading if/else, or if we'd be better off doing a `match` here. Either way, the `get_otel_attribute_value_as_string_from_resource_or_span` could return an option here and you could just match on `Some` to make these easier.

Reply:
I could break it down to smaller pieces, yet I don't think a match would be too readable here to accomplish what you expect on a `match`, as I'd have to extract the matching pattern outside to then execute the method to call the resource from the right key